Sources/UnidirectionalBinding.swift

```diff
@@ -11,167 +11,61 @@ precedencegroup BindingPrecedence {
-	/// ...
-	/// // Terminates binding before property dealloc or
-	/// // signal's `completed` event.
-	/// disposable.dispose()
-	/// ````
-	///
-	/// - parameters:
-	///   - target: A target to be bond to.
-	///   - producer: A producer to bind.
-	///
-	/// - returns: A disposable that can be used to terminate binding before the
-	///            deinitialization of the target or the producer's `completed
-	///            event.
+extension SignalProducer: BindingSourceProtocol {
 	@discardableResult
-	public static func <~ <Source: SignalProducerProtocol>(target: Self, producer: Source) -> Disposable where Source.Value == Value, Source.Error == NoError {
+	public func observe(_ observer: Observer<Value, Error>, lifetime: Lifetime) -> Disposable? {
 		var disposable: Disposable!
 
-		producer
-			.take(during: target.lifetime)
+		take(during: lifetime)
 			.startWithSignal { signal, signalDisposable in
 				disposable = signalDisposable
-				target <~ signal
+				signal.observe(observer)
 			}
 
 		return disposable
 	}
+}
 
-	/// Binds a property to a target, updating the target's value to the latest
-	/// value sent by the property.
-	///
-	/// - note: The binding will automatically terminate when either the target or
-	///         the property deinitializes.
-	///
-	/// ````
-	/// let dstProperty = MutableProperty(0)
-	/// let srcProperty = ConstantProperty(10)
-	/// dstProperty <~ srcProperty
-	/// print(dstProperty.value) // prints 10
-	/// ````
-	///
-	/// ````
-	/// let dstProperty = MutableProperty(0)
-	/// let srcProperty = ConstantProperty(10)
-	/// let disposable = (dstProperty <~ srcProperty)
-	/// ...
-	/// disposable.dispose() // terminate the binding earlier if
-	///                      // needed
-	/// ````
-	///
-	/// - parameters:
-	///   - target: A target to be bond to.
-	///   - property: A property to bind.
-	///
-	/// - returns: A disposable that can be used to terminate binding before the
-	///            deinitialization of the target or the source property.
+extension Property: BindingSourceProtocol {
 	@discardableResult
-	public static func <~ <Source: PropertyProtocol>(target: Self, property: Source) -> Disposable where Source.Value == Value {
-		return target <~ property.producer
+	public func observe(_ observer: Observer<Value, NoError>, lifetime: Lifetime) -> Disposable? {
+		return producer.observe(observer, lifetime: lifetime)
 	}
 }
 
+/// Describes a target which can be bound.
+public protocol BindingTargetProtocol: class {
+	associatedtype Value
+
+	/// The lifetime of `self`. The binding operators use this to determine when
+	/// the binding should be teared down.
+	var lifetime: Lifetime { get }
+
+	/// Consume a value from the binding.
+	func consume(_ value: Value)
+}
+
+@discardableResult
+public func <~ <Target: BindingTargetProtocol, Source: BindingSourceProtocol>(target: Target, source: Source) -> Disposable? where Source.Value == Target.Value, Source.Error == NoError {
```

Comment:
Since it's generic over both sides, I think it makes sense to define it at the top level.